src/workerd/io/worker-fs.h

```diff
@@ -586,6 +588,13 @@ class VirtualFileSystem {
   // API and posix in general.
   virtual void closeFd(jsg::Lock& js, int fd) const = 0;
 
+  // Returns an opaque RAII handle that wraps a file descriptor.
+  // When it is dropped, the file descriptor will be closed. The handle
+  // holds a weak reference to the VirtualFileSystem so that, on the off
+  // chance the VirtualFileSystem is destroyed while the handle is still
+  // alive, destruction of the handle will become a no-op.
+  virtual kj::Own<void> wrapFd(jsg::Lock& js, int fd) const KJ_WARN_UNUSED_RESULT = 0;
```

Comment:
Is the `Own<void>` pattern idiomatic KJ for opaque pointers? What is the benefit of making this opaque?

Reply:
Yeah we use this pattern in a number of places and I would consider it idiomatic for kj. The benefit is really just reducing the external API surface area when it's not needed. In this case I want it to be clear that this is purely an RAII handle. The only thing you can do with it is hold it then drop it.